verifier/src/main/java/com/aws/jverify/verifier/JavaToDafnyCompiler.java

```diff
@@ -435,51 +419,81 @@ private Statement translateStatement(JCTree.JCStatement statement) {
         throw new NotImplementedException(statement.getClass().getName());
     }
 
-    private boolean findWhileHeaderStatement(JCTree.JCStatement statement,
-                                             List<AttributedExpression> invariants,
-                                             List<Expression> decreases, 
-                                             List<FrameExpression> modifies) {
-        if (!(statement instanceof JCTree.JCExpressionStatement expressionStatement)) {
-            return false;
-        }
-        var expr = expressionStatement.getExpression();
-        if (!(expr instanceof JCTree.JCMethodInvocation invocation)) {
-            return false;
-        }
+    private void checkEmptyExpressions(List<AttributedExpression> expressions, 
+                                       String typeName, 
+                                       String containerName) {
+      for(var expr : expressions) {
+          throw new RuntimeException(typeName + " are not allowed in a " + containerName);
+      }
```

Comment:
tiny nitpick - I feel the intent is clearer this way, what do you think? (I initially thought you might've intended to check something about `expr` until I looked at the call sites.)

```suggestion
      if (!expressions.isEmpty()) {
          throw new RuntimeException(typeName + " are not allowed in a " + containerName);
      }
```

Reply:
In the future I expect we won't throw an exception here but report an error, so then it can report multiple errors, that's why I left the loop here.